server/routes/requests.ts

```diff
@@ -0,0 +1,192 @@
+
+const router = express.Router();
+
+module.exports = router;
+const determineMessageStatus: (arg1, arg2, arg3) => [string, number] = (request, accepted, approved) => {
+    switch (true) {
+        case !request:
+            return ['Swap request not found', 404];
+        case !accepted && !approved:
+            return ['', 202];
+        default:
+            return ['Invalid Request', 400];
+    }
+};
+
+const postNotification = (requestId, userId, notification, message, authorization) =>
+    axios.post(
+        `${process.env.SECRET_IP}api/notifications`,
+        { requestId, userId, notification, message },
+        { headers: { authorization } },
+    );
+
+router.get('/requests', certify, async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const loggedInId: number = determineLoginId(req.headers.authorization);
+        const requests = await db.Request.findAll({
+            where: { userId: loggedInId, approved: false },
+            attributes: [
+                ['id', 'requestId'],
+                'requesteeUserId',
+                'type',
+                'accepted',
+                'approved',
+                'createdAt',
+                'taskId',
+                'message',
+                'replace',
+            ],
+        });
+        return requests.length > 0
+            ? res.status(200).json(requests)
+            : res.status(404).send({ message: 'Swap requests not found' });
+    } catch (err) {
+        next(err);
+        return res.status(503).send({ message: 'Server error, try again later' });
+    }
+});
+
+router.get('/requests/:requestId', certify, async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const { requestId: id } = req.params;
+        const request = await db.Request.findOne({
+            where: { id },
+            attributes: ['id', 'requesteeUserId', 'type', 'accepted', 'taskId'],
+        });
+        return request ? res.json(request) : res.status(404).send({ message: 'Not Found' });
+    } catch (err) {
+        next(err);
+        return res.status(503).send({ message: 'Server error, try again later' });
+    }
+});
+
+router.post('/requests', certify, async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const { yourTaskId, myTaskId, message } = req.body;
+        const { authorization } = req.headers;
+
+        const yourTask = await db.Task.findOne({ where: { id: yourTaskId } });
+        const { userId: yourUserId } = await db.UserRole.findOne({ where: { id: yourTask.userRoleId } });
+
+        const myTask = await db.Task.findOne({ where: { id: myTaskId } });
+        const { userId: myUserId } = await db.UserRole.findOne({ where: { id: myTask.userRoleId } });
+
+        if (!myTask) res.status(404).send({ message: 'Task not found' });
+        await myTask.update({ status: 'changeRequested' });
+
+        const request = await db.Request.create({
+            requesteeUserId: yourUserId,
```

Comment:
i personally found the `my`s and `your`s a little ambiguous until I read this line. We could make it more obvious/boring with 
```suggestion
            requesteeUserId,
```
and then replace `your...` -> `requestee...` and `my...` -> `requester...` or `requesting...` (unless I have the POV backwards)

Reply:
haha very interesting, I changed it from requestee/requester to your/my because I thought it was clearer (also less wordy and thus cleaner), but as I'm hearing two voices say that requestee makes more sense, then I think that's the way to go.